### PR TITLE
Add DuckDB Scan

### PR DESCRIPTION
**Pull request by karansampath**
Adds a scan for DuckDB, allowing for the following to work:

```python
context.read.duckdb(db_url, table_name).do_stuff()
```

Also cleans up some code from https://github.com/aryn-ai/sycamore/pull/480, and does some refactoring to allow for better organization with the new connector structure.

### CODE REVIEW

File: lib/sycamore/sycamore/connectors/duckdb/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from sycamore.connectors.duckdb.duckdb_scan import DuckDBScan
+
+__all__ = [
+    "DuckDBScan",
+]
```
Comment (HenryL27) [inline lib/sycamore/sycamore/connectors/duckdb/__init__.py:5]: put the writer stuff in here too

**Review by HenryL27 — COMMENTED** [review]
One code comment. Not sure I agree with moving writers/* to connectors/writers/*.

File: lib/sycamore/sycamore/tests/unit/writers/test_base.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from pathlib import Path
 from sycamore.data.document import Document, MetadataDocument
 from sycamore.plan_nodes import Node
-from sycamore.writers.base import BaseDBWriter
+from sycamore.connectors.writers.base import BaseDBWriter
```
Comment (HenryL27) [inline lib/sycamore/sycamore/tests/unit/writers/test_base.py:5]: ok I'm on board with the 'everything to connectors'. I'm gonna make you move this file to test/unit/connectors/writers/whatever to reflect that tho

**Review by HenryL27 — APPROVED** [review]
one move to make

File: lib/sycamore/sycamore/tests/unit/writers/test_base.py (hunk 2 above)
Comment (karansampath) [reply lib/sycamore/sycamore/tests/unit/writers/test_base.py:5]: Fair enough, done.

**Pull request merged**